### PR TITLE
Collectuserdata

### PR DESCRIPTION
**Pull request by RamonPeek**
## Description

Added an endpoint for collecting all userdata within DeX. For now I have created a custom model which only contains the user-object. In the future this may be extended with (for example) data we use based on the user's behaviour on the platform.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I did not update API Controllers, if I did, I added/changed Postman tests
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] I updated the changelog with an end-user readable description
-   [ ] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Access the endpoint [GET] /api/user/{userId}/allData
2. No tests are required as only services are used which are already fully tested.
3. When providing a valid userId it should return all data of the user; if a non-existing userId is provided, it will return a 404.

## https://github.com/DigitalExcellence/dex-backend/issues/77

Closes: #77

### CODE REVIEW

**Comment by Brend-Smits** [comment]
Test are failing

File: API/Controllers/UserController.cs [hunk 1]
```diff
@@ -39,15 +40,17 @@ public class UserController : ControllerBase
 
         private readonly IMapper mapper;
         private readonly IUserService userService;
+        private readonly IProjectService projectService;
 
         /// <summary>
         ///     Initialize a new instance of UserController
         /// </summary>
         /// <param name="userService"></param>
         /// <param name="mapper"></param>
```
Comment (jcoenen96) [inline API/Controllers/UserController.cs:49]: Parameter 'projectService' has no matching param tag in the XML comment for 'UserController.UserController(IUserService, IProjectService, IMapper)' (but other parameters do)

File: Models/UserData.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Models
+{
+    //This class will extend all future data regarding users, apart from the user-object itself.
+    public class UserData
+    {
+        public User User { get; set; }
+
```
Comment (jcoenen96) [inline Models/UserData.cs:28]: I would say that all the projects created by the user are part of his/her userdata

Comment (RamonPeek) [reply Models/UserData.cs:28]: These are present in the user object itself and will thus be retrieved or deleted aswell. The userdata might contain stuff about userpreferences etc in the future

**Pull request closed without merging**